packages/datadog_flutter_plugin/android/src/test/kotlin/com/datadoghq/flutter/DatadogSdkPluginTest.kt

```diff
@@ -693,6 +704,211 @@ class DatadogSdkPluginTest {
         verify(mockResult).success(null)
     }
 
+    @Test
+    fun `M clear user info W called through MethodChannel`(
+        forge: Forge
+    ) {
+        // GIVEN
+        val config = mapOf(
+            "clientToken" to forge.aString(),
+            "env" to forge.anAlphaNumericalString()
+        )
+        plugin.initialize(config, TrackingConsent.GRANTED)
+
+        val id = forge.aString()
+        val name = forge.aNullable { forge.aString() }
+        val email = forge.aNullable { forge.aString() }
+        val extraInfo = forge.exhaustiveAttributes()
+        val firstMockResult = mock<MethodChannel.Result>()
+        plugin.onMethodCall(MethodCall(
+            "setUserInfo",
+            mapOf(
+                "id" to id,
+                "name" to name,
+                "email" to email,
+                "extraInfo" to extraInfo
+            )
+        ), firstMockResult)
+
+        // WHEN
+        val methodCall = MethodCall(
+            "clearUserInfo",
+            mapOf<String, Any?>()
+        )
+        val mockResult = mock<MethodChannel.Result>()
+        plugin.onMethodCall(methodCall, mockResult)
+
+        // THEN
+        val core = Datadog.getInstance()
+        val coreFeature: Any = core.getPrivate("coreFeature")!!
+        // Ensure safe write before attempting to read
+        Thread.sleep(10)
+        val userInfo: UserInfo? = coreFeature
+            .getFieldValue<Any, Any>("userInfoProvider")
+            .getFieldValue("internalUserInfo")
```

Comment:
alternatively you can cast core to `InternalSdkCore` and call `getDatadogContext()` to avoid reflection and flaky `Thread.sleep`

Reply:
When attempting to do that I get an exception trying to call `getSafe`

> java.lang.NullPointerException: Cannot invoke "android.content.res.Resources.getConfiguration()" because the return value of "android.content.Context.getResources()" is null

Is there a way around that?

Reply:
Probably then you need to mock `Context` you pass to the SDK initialization. But if it is too complex, you can use reflection (although it imposes using `Thread.sleep`).